src/main/java/bio/terra/workspace/service/datareference/model/ReferenceObject.java

```diff
@@ -11,19 +10,28 @@
  * rules. ReferenceObject implementations must be fully serializable by the output of getProperties.
  */
 public interface ReferenceObject {
-  Map<String, String> getProperties();
 
   /**
-   * Convenience method for building a concrete referenceObject from a map and type.
+   * Method for serializing a ReferenceObject to a json string.
    *
-   * <p>This is used for deserializing ReferenceObjects from their properties and type.
+   * <p>This method is used for Stairway serialization. Database serialization should use the static
+   * {@Code DataReferenceDao.serializeReferenceObject} method instead.
    */
-  static ReferenceObject toReferenceObject(DataReferenceType type, Map<String, String> properties) {
+  String toJson();
+
+  /**
+   * Method for deserializing a ReferenceObject from a json string and its type.
+   *
+   * <p>This method is used for Stairway deserialization. Database deserialization should use the
+   * static {@Code DataReferenceDao.deserializeReferenceObject} method instead.
```

Comment:
I'm not totally convinced we need two serialization formats (dao & stairway) yet. One day, I could see us wanting to store different information in the DAO than in the stairway flight maps, but I'm not sure what feature would cause us to split. When we do split, we can split the code and do backwards compatibility support as needed, no different than today.

So far the only methods on this interface are that it supports json serialization that we need backwards compatibility for. It doesn't seem crazy to rely on that in the DAO as well as for Stairway.

At least, it's worth explaining why we need two json serialization formats for SnapshotReferences/ReferenceObjects.

Reply:
Yeah, this was something I was less sure about. I think there's a slightly higher risk of making backwards-incompatible changes if we use the same serialization in both places, but it's a pretty small risk and this way takes a lot of boilerplate.  If you're fine using the same serialization for both, I'll go with that.

Reply:
Yeah. I think we should mitigate the risk of backwards incompatible changes with unit tests commenting about what failures indicate. With that, for now it seems fine to me to write the serialization once.